### PR TITLE
Bugfix/pptt 3435 remove auth token

### PR DESCRIPTION
**Pull request by prk3**
Please review commits one by one. Commit messages explain what has happened.

### CODE REVIEW

File: data_collection/api/tracker_debugger_api.yaml [hunk 1]
```diff
@@ -3346,10 +3346,6 @@ components:
           description: '[Custom variables](https://help.piwik.pro/analytics/custom-variables/) set on the ``page`` scope.'
           type: string
           example: '{"1":["AspectRatio","16:9"],"2":["Height","1080"],"3":["Width","1920"]}'
-        bots:
```
Comment (azarecki) [inline data_collection/api/tracker_debugger_api.yaml:3349]: There is `bots` parameter in the Collector Front, no in the Processor. We forget about it or removed intentionally?

File: data_collection/api/tracking_api.yaml [hunk 2]
```diff
@@ -1,62 +1,62 @@
...
-          #TODO: Confirm that those are only valid options.
-          minimum: 0
-          maximum: 1
+          description: |
+            **Recommended** Whether the request should be recorded.
+
+            *IMPORTANT*: Requests should contain rec=1 parameter, otherwise they won't appear in reports.
+          schema:
+            type: integer
+            #TODO: Confirm that those are only valid options.
```
Comment (azarecki) [inline data_collection/api/tracking_api.yaml:40]: Yes, these two are only valid, we can remove the todo

File: data_collection/api/tracking_api.yaml [hunk 3]
```diff
@@ -296,92 +338,102 @@ paths:
...
+            minimum: 6
+            maximum: 6
           example: ff5c6d
         - name: idgoal
           in: query
-          #TODO: Unify Ecommerce naming scheme. Rest of parameters refer to Ecommerce order, not cart.
+          #TODO: Unify E-commerce naming scheme. Rest of parameters refer to E-commerce order, not cart.
           description: |
-            Goal ID. Signifies that a goal has been reached and tracks its conversion.
+            ID of the converted goal. Value `0` indicates that it's an E-commerce event.
```
Comment (azarecki) [inline data_collection/api/tracking_api.yaml:351]: Looks like information about `0` is repeated

File: data_collection/api/tracking_api.yaml [hunk 4]
```diff
@@ -390,150 +442,170 @@ paths:
...
         - name: e_c
           in: query
           description: Custom event category.
-          type: string
-          minLength: 1
-          #TODO: Specify maximal allowed length.
+          schema:
+            type: string
+            minLength: 1
+            #TODO: Specify maximal allowed length.
```
Comment (azarecki) [inline data_collection/api/tracking_api.yaml:550]: We can do it now, right?

File: data_collection/api/tracking_api.yaml [hunk 5]
```diff
@@ -248,46 +265,71 @@ paths:
...
+            type: string
           example: sherlock holmes
         - name: search_cat
           in: query
+          deprecated: true
           description: |
-            Internal search category. This is used to track if visitor specified a category during a search of the application.
-            See: [Site search](https://help.piwik.pro/analytics/site-search/).
+            Internal search category. Used for tracking category the visitor specified during a search in the application.
+            See: [Site search](https://help.piwik.pro/support/reports/internal-search-report/).
```
Comment (RomanSek) [inline data_collection/api/tracking_api.yaml:301]: Do we need to update this link in other versions?

File: data_collection/web/javascript_tracking_client/api.rst [hunk 6]
```diff
@@ -1143,14 +1143,14 @@ Download and Outlink
 
 .. _jtc-api-enableLinkTracking:
 
-.. function:: enableLinkTracking(enable)
+.. function:: enableLinkTracking([trackMiddleAndRightClicks])
```
[the remark this answers is not in the document]
Comment (RomanSek) [reply data_collection/web/javascript_tracking_client/api.rst:1146]: We didn't use this convention anywhere else, but we can think about introducing it. Usually optional parameters have a default value in parameter description.

**Review by RomanSek — APPROVED** [review]

File: data_collection/web/javascript_tracking_client/api.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (azarecki) [reply data_collection/web/javascript_tracking_client/api.rst:1146]: Ok, so let's do not change it here, we can add it with a separate task (if we want to). Thanks

File: data_collection/api/tracking_api.yaml (hunk 3 above)
Comment (prk3) [reply data_collection/api/tracking_api.yaml:351]: Yes, I will restore the old description.

File: data_collection/api/tracking_api.yaml (hunk 5 above)
Comment (prk3) [reply data_collection/api/tracking_api.yaml:301]: What other versions do you have in mind? From anonymized log endpoint?

File: data_collection/api/tracking_api.yaml (hunk 4 above)
Comment (prk3) [reply data_collection/api/tracking_api.yaml:550]: Ok, I will add it.

File: data_collection/api/tracker_debugger_api.yaml (hunk 1 above)
Comment (prk3) [reply data_collection/api/tracker_debugger_api.yaml:3349]: We had it in processor but it wasn't used.

**Comment by RomanSek** [comment]
I created [preview](https://developers.piwik.pro/en/bugfix-pptt-3435-remove-auth-token/data_collection/api/tracker_debugger_api.html) if you want.

**Review by wronan — APPROVED** [review]

**Review by azarecki — APPROVED** [review]

**Pull request merged**